packages/signature-controller/src/types.ts

```diff
@@ -71,11 +77,29 @@ export type MessageParamsTyped = MessageParams & {
         primaryType: string;
         message: Json;
       };
-
+  /** Signature method V1, V3 or V4 */
+  signatureMethod?: string;
   /** Version of the signTypedData request. */
   version?: string;
 };
 
+export type DecodedRequestInfo =
+  | {
+      assetType: string;
+      changeType: string;
+      address: string;
+      amount: string;
+      contractAddress: string;
+    }
+  | {
```

Comment:
Rather than defining these as a union of alternate types, could we define additional types such as `DecodingError` and move them into nested properties such as `data` and `error`?